### PR TITLE
lower various trig ops to math lib

### PR DESCRIPTION
**Pull request by caoimhinuibrian**
Signed-off-by: Kevin O'Brien <caomhin@us.ibm.com>

### CODE REVIEW

**Comment by caoimhinuibrian** [comment]
atan was being lowered to an mlir intrinsic, but there is no lowering of that to LLVM and the discussion on the MLIR channel seems to say there is no plan to add the lowering ... so, I just lowered it directly to an LLVM call like all the others.

File: test/backend/test.py [hunk 1]
```diff
@@ -125,14 +127,17 @@
     # Argmin
 
     # Asin
+    "test_asin_cpu": (test_static_dynamic,),
 
     # Asinh
+    "test_asinh_cpu": (test_static_dynamic,),
 
     # Atan
     #"test_atan_example_cpu": (test_static_dynamic,),
```
Comment (AlexandreEichenberger) [inline test/backend/test.py:136]: Are the _example test not working? If they are, I would enable them too

File: src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp [hunk 2]
```diff
@@ -531,7 +531,84 @@ struct MathFunctionName {
 
 template <>
 struct MathFunctionName<KrnlErfOp> {
-  static std::string functionName() { return "erff"; }
+  static std::string functionName(mlir::Type type) { return "erff"; }
```
Comment (AlexandreEichenberger) [inline src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp:534]: I would use the same template as you used, asserting an error if type is not float. If the erf is used for double format, that would be great too.

File: src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp [hunk 3]
```diff
@@ -531,7 +531,84 @@ struct MathFunctionName {
...
+};
+
+template <>
+struct MathFunctionName<KrnlAcosOp> {
+  static std::string functionName(mlir::Type type) {
+    if (type.isF32())
+      return "acosf";
+    if (type.isF64())
+      return "acos";
+    assert(false && "Unsupported type for acos");
```
Comment (AlexandreEichenberger) [inline src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp:544]: The proper form for `assert(false)`  is `llvm_unreachable("message")`.

**Review by AlexandreEichenberger — COMMENTED** [review]
Great, couple of minor changes requested.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, thanks for the changes, adding the test, and supporting ERF double.

**Comment by AlexandreEichenberger** [comment]
Looks like there is still a formatting error, otherwise all good.

**Pull request merged**